conans/client/cmake.py

```diff
@@ -71,6 +72,33 @@ def _generator(self):
 
         raise ConanException("Unknown cmake generator for these settings")
 
+    def _cmake_compiler_options(self, target_os, target_arch, target_comp):
+        host_platform = platform.system()
+        same_platform = (target_os == host_platform)
+
+        cmake_flags = []
+        cmake_flags.append("-DCMAKE_SYSTEM_NAME=%s" % target_os)
+
+        if target_comp == "clang":
+            cmake_flags.append("-DCMAKE_C_COMPILER=clang")
+            cmake_flags.append("-DCMAKE_CXX_COMPILER=clang++")
+        elif target_comp == "gcc":
+            if target_os == "Windows" and not same_platform:
+                if target_arch == "x86":
+                    cmake_flags.append("-DCMAKE_C_COMPILER=i686-w64-mingw32-gcc")
+                    cmake_flags.append("-DCMAKE_CXX_COMPILER=i686-w64-mingw32-g++")
+                else:
+                    cmake_flags.append("-DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc")
+                    cmake_flags.append("-DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++")
+            else:
+                cmake_flags.append("-DCMAKE_C_COMPILER=gcc")
```

Comment:
I have also strong concerns now for these two lines. I have been testing in Win/MinGW, and the problem is that those values overwrite the existing environment variables $CC and $CXX.
So far, the way that users have to control the compiler they are using, is with those environment variables. For other cases ("clang"), it was just experimental, very few people using clang in Windows. But I think that the ``CMake`` helper changing this behavior could badly affect existing users projects.